tsl/test/expected/hypercore_ddl.out

```diff
@@ -53,10 +48,144 @@ select chunk, amname from chunk_info where hypertable = 'readings'::regclass;
  _timescaledb_internal._hyper_1_4_chunk | hypercore
 (4 rows)
 
--- Pick a chunk to truncate that is not the first chunk. This is
+----------------------------------------------------------------
+-- Test ALTER TABLE .... ALTER COLUMN commands
+-- This should fail since "location" is NOT NULL
+\set ON_ERROR_STOP 0
+insert into readings(time,device,temp,humidity,jdata)
+values ('2024-01-01 00:00:00', 1, 99.0, 99.0, '{"magic": "yes"}'::jsonb);
+ERROR:  null value in column "location" of relation "_hyper_1_9_chunk" violates not-null constraint
+\set ON_ERROR_STOP 1
+-- Test altering column definitions to drop NOT NULL
+alter table readings alter column location drop not null;
+-- This should now work since we allow NULL values
+insert into readings(time,device,temp,humidity,jdata)
+values ('2024-01-01 00:00:00', 1, 99.0, 99.0, '{"magic": "yes"}'::jsonb);
+select count(*) from readings where location is null;
+ count 
+-------
+     1
+(1 row)
+
+select compress_chunk(show_chunks('readings'), hypercore_use_access_method => true);
+NOTICE:  chunk "_hyper_1_1_chunk" is already compressed
+NOTICE:  chunk "_hyper_1_2_chunk" is already compressed
+NOTICE:  chunk "_hyper_1_3_chunk" is already compressed
+NOTICE:  chunk "_hyper_1_4_chunk" is already compressed
+             compress_chunk              
+-----------------------------------------
+ _timescaledb_internal._hyper_1_1_chunk
+ _timescaledb_internal._hyper_1_2_chunk
+ _timescaledb_internal._hyper_1_3_chunk
+ _timescaledb_internal._hyper_1_4_chunk
+ _timescaledb_internal._hyper_1_10_chunk
+(5 rows)
+
+select count(*) from readings where location is null;
+ count 
+-------
+     1
+(1 row)
+
+-- Pick a chunk to play with that is not the first chunk. This is
 -- mostly a precaution to make sure that there is no bias towards the
 -- first chunk and we could just as well pick the first chunk.
 select chunk from show_chunks('readings') x(chunk) limit 1 offset 3 \gset
+-- We should not be able to set the not null before we have removed
+-- the null rows in the table. This works for hypercore-compressed
+-- chunks but not for heap-compressed chunks.
+\set ON_ERROR_STOP 0
+alter table readings alter column location set not null;
+ERROR:  column "location" of relation "_hyper_1_10_chunk" contains null values
```

Comment:
We might want to check that this works for rows that are both in non-compressed and compressed relations.

Reply:
Added a row with nulls after compressing all chunks.